src/platform/packages/shared/kbn-esql-validation-autocomplete/src/shared/resources_helpers.ts

```diff
@@ -90,9 +90,26 @@ export class QueryColumns {
     if (!this.fullQueryCacheKey) return;
 
     const getFields = async (queryToES: string) => {
+      const freshColumns = await this.resourceRetriever?.getColumnsFor?.({
```

Comment:
Here's the detail. We move getColumnFor before checking the cache. This means we always call it. The thing I didn't like was that the getColumnFor implementation contains potential HTTP requests, but with memoization we should be fine.